core-api/src/main/java/com/optimizely/ab/config/parser/ConditionJacksonDeserializer.java

```diff
@@ -48,20 +48,37 @@ public ConditionJacksonDeserializer() {
     @Override
     public Condition deserialize(JsonParser parser, DeserializationContext context) throws IOException {
         JsonNode node = parser.getCodec().readTree(parser);
-
-        String conditionsJson = node.textValue();
-        JsonNode conditionsTree = objectMapper.readTree(conditionsJson);
+        JsonNode conditionsTree = node;
+        if (node.isTextual()) {
+            String conditionsJson = node.textValue();
+            conditionsTree = objectMapper.readTree(conditionsJson);
+        }
         Condition conditions = ConditionJacksonDeserializer.parseConditions(objectMapper, conditionsTree);
 
         return conditions;
     }
 
     protected static Condition parseConditions(ObjectMapper objectMapper, JsonNode conditionNode) throws JsonProcessingException {
+
+        if (conditionNode.size() == 0) {
+            return new EmptyCondition();
```

Comment:
You could also just construct an OrCondition with no children, since that's semantically what an empty array represents.

(Applies to ConditionUtils.java too)

Reply:
But, ["or"] should be null where as [] should be true.